modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/SampleStreamInterceptor.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.synapse.transport.passthru;
+
+import org.apache.axis2.context.MessageContext;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+
+/**
+ * Sample stream interceptor, which intercepts source and target request to print the stream passing via the engine.
+ * <pre>
+ * {@code
+ * <interceptors xmlns:svns="http://org.wso2.securevault/configuration">
+ *     <interceptor class="org.apache.synapse.transport.passthru.SampleStreamInterceptor">
+ *         <parameter name="charset" value="ISO-8859-1"/>
+ *         <parameter name="enableInterception" value="true"/>
+ *     </interceptor>
+ * </interceptors>
+ * }
+ * </pre>
+ */
+public class SampleStreamInterceptor extends DefaultStreamInterceptor {
```

Comment:
Why don't we give it a proper name, such as 'LoggingStreamInterceptor' or something? Because, I feel like we will start asking customers to add this to debug issues in support in the near future. :)

Reply:
updated in https://github.com/wso2/wso2-synapse/pull/1695